### PR TITLE
Updated .waitFor() to allow passing store instances instead of tokens. 

### PR DESCRIPTION
**Pull request by mattmccray**
Will also accept multiple params instead of only an array of items. Backwards compatible with Facebook's version of waitFor().

As discussed in #38

### CODE REVIEW

File: src/alt.js [hunk 1]
```diff
@@ -176,11 +176,21 @@ const StoreMixin = {
...
     }
-    tokens = Array.isArray(tokens) ? tokens : [tokens]
+
+    if (arguments.length === 1) {
+      sources = Array.isArray(sources) ? sources : [sources]
+    } else {
+      sources = Array.prototype.slice.call( arguments)
+    }
+
+    let tokens = sources.map(( source)=>{
```
Comment (goatslacker) [inline src/alt.js]: Just a few whitespace nits:

removing leading whitespace from open paren, and add some whitespace between arrow and paren/brace:

`let tokens = sources.map((source) => {`

same for the slice call above.

File: test/index.js [hunk 2]
```diff
@@ -208,6 +209,19 @@ class LifeCycleStore {
...
 let lifecycleStore = alt.createStore(LifeCycleStore)
 
+class ThirdStore {
+  constructor() {
+    this.bindAction(myActions.updateName, this.onUpdateName)
+  }
+
+  onUpdateName() {
+    this.waitFor( myStore, secondStore) // Not referencing dispatchToken!
+    this.name= secondStore.getState().name +'3'
```
Comment (goatslacker) [inline test/index.js]: whitespace around operators

**Comment by goatslacker** [comment]
lgtm!

**Comment by mattmccray** [comment]
I think this addresses the whitespace niggles.

You might consider putting a style guide of some sort in the repo (although it's possible I just missed it). :grin:

**Comment by goatslacker** [comment]
#25 working on it! :)

File: src/alt.js [hunk 3]
```diff
@@ -176,11 +176,21 @@ const StoreMixin = {
     })
   },
 
-  waitFor(tokens) {
-    if (!tokens) {
+  waitFor(sources) {
+    if (! sources) {
```
Comment (goatslacker) [inline src/alt.js]: missed one!

**Comment by mattmccray** [comment]
OK. Maybe that's the last one. :)

**Comment by goatslacker** [comment]
I'll update the docs. Thanks :smile:

**Comment by goatslacker** [comment]
ced88f87f5f48de7cc58e1ad642780509ce6bec2

**Pull request merged**